src/vs/workbench/contrib/search/browser/searchView.ts

```diff
@@ -1757,6 +1788,28 @@ export class SearchView extends ViewletPane {
 		super.saveState();
 	}
 
+	private async retrieveFileStats(): Promise<void> {
+		for (const fileMatch of this.searchResult.matches()) {
+			if (!fileMatch.fileStat) {
+				await fileMatch.resolveFileStat(this.fileService);
+			}
+		}
+		return Promise.resolve(undefined);
+	}
+
+	private async updateFileStats(elements: FileMatch[]): Promise<void> {
+		for (const fileMatch of elements) {
+			await fileMatch.resolveFileStat(this.fileService);
```

Comment:
This will check each file syncronously. You might consider using Promise.all to run these in parallel.

Reply:
Thanks, I was going to look into that